app/controllers/hubstats/application_controller.rb

```diff
@@ -5,9 +5,14 @@ class ApplicationController < ApplicationController
 
     private
     def set_time
-      cookies[:hubstats_index] ||= 2
-      @start_date = DATE_RANGE_ARRAY[cookies[:hubstats_index].to_i][:date].ago.to_date
-      @end_date = Time.now
+      if cookies[:hubstats_index] == "null~~null" || cookies[:hubstats_index] == nil
+        times = "#{Date.today - 14}~~#{Date.today}"
+      else
+        times = cookies[:hubstats_index]
+      end
+      
+      @start_date = times.split("~~").first.to_date
+      @end_date = times.split("~~").last.to_date + 1
```

Comment:
I assume `to_date` throws an exception if the string value is an un-parsable date value which we should code for using a rescue block.

You are interpolating a cooking string just to split it.  Your default case should use the date values directly to save CPU cycles.